cmake/Dependencies.cmake

```diff
@@ -23,8 +35,47 @@ list(APPEND Caffe_LINKER_LIBS ${GFLAGS_LIBRARIES})
 # ---[ Google-protobuf
 include(cmake/ProtoBuf.cmake)
 
+if( MSVC AND NOT WITH_PREBUILD_MSVC_DEPENDENCIES )
```

Comment:
ZLIB and SZIP are not dependencies of Caffe per se but transitive dependencies of HDF5. This should be handled in a HDF5Config.cmake file or a custom FindHDF5.cmake module.